account_invoice_rounding/account.py

```diff
@@ -174,8 +174,9 @@ def _compute_amount(self):
                 if 'amount_tax' in swedish_rounding:
                     self.amount_tax = swedish_rounding['amount_tax']
                 elif 'amount_untaxed' in swedish_rounding:
-                    self.amount_untaxed = (
-                        swedish_rounding['amount_untaxed'])
+                    if sum(line.price_subtotal for line in self.invoice_line) != self.amount_untaxed:
+                        self.amount_untaxed = (
+                            swedish_rounding['amount_untaxed'])
```

Comment:
So basically swedish_rounding['amount_untaxed'] is wrong if there is an adjusting line.

There was an issue solved in version 9.0 that implied two write due to method call cycling:
https://github.com/OCA/account-invoicing/commit/be97dfa9248ebcd0f0889e6da999e23a77ce824a#diff-9096b1ac1c24a29588050b4c5a6d3726

The value in swedish_rounding['amount_untaxed'] comes from here:
https://github.com/OCA/account-invoicing/blob/8.0/account_invoice_rounding/account.py#L58

Thus we should probably fix it here. Still this value is set twice and that's a bad idea. I'm doubtful about the first write in `self.amount_untaxed` on line 168. It could probably be moved after the `if swedish_rounding:` in a elif to set it as we probably still want to remove it from untaxed amount.

Could you give a try to backport the tests made in later version of this module to make it stable:

https://github.com/OCA/account-invoicing/blob/10.0/account_invoice_rounding/tests/test_invoice_rounding.py

Reply:
The test was added by #392. Until the fix is applied, the test will fail.